src/components/chat/chat.js

```diff
@@ -122,8 +125,28 @@ class Chat extends React.Component {
   }
 
   hangUp = () => {
-    console.log('sent to BE')
     this.props.socket.emit('hang up', {roomId: this.state.roomId});
+    this.setState({
+      showModal: true
+    })
+  }
+
+  showEndChatModal = () => {
+    if (this.state.showModal) {
+      return <ModalEndChat closeChatModal={this.closeChatModal} history={this.props.history} questionId={this.state.questionId} tutorOrLearner={this.state.tutorOrLearner}/>
+    }
+  }
+
+  openChatModal = () => {
+    this.setState({
+      showModal: true
+    })
+  }
+
+  closeChatModal = () => {
+    this.setState({
+      showModal: false
+    })
```

Comment:
Instead of writing two seperate functions, should we  make a single function called `toggleChatModal `that setStates to `showModal: !this.state.showModal`?